[Site creation] Style SiteSegmentsWizardContent

ctarda opened the pull request:
Fixes #10483 

A round of polishing to SiteSegmentsWizardContent by:
- [ ] Update the cells to match the styles in Zeplin. The second line of the cell (subtitle) should overflow to two lines if the content is large enough. Title and subtitle, as a block, should be centred vertically. The icon should be smaller.
- [ ] Prevent the header from being hidden by the top navigation bar

Before:
![simulator screen shot - iphone x - 2018-11-19 at 13 56 57](https://user-images.githubusercontent.com/2722505/48689024-a7425780-ec03-11e8-9673-fde5968debc9.png)

After:
![simulator screen shot - iphone x - 2018-11-19 at 14 12 17](https://user-images.githubusercontent.com/2722505/48689370-313ef000-ec05-11e8-9519-8f8df52f637c.png)

Please note: 
- the table view header still overflows over the table view. This will be fixed in an upcoming PR (story #10479)
- as part of the diff, there is an update to the mock data (`SiteSegmentsService`). This service, like the rest of mocks will be removed before release

To test:
- Checkout the branch, run the tests.
- Navigate to My Sites > Tap + > Enhanced site creation

kwonye commented on WordPress/Classes/Services/SiteSegmentsService.swift:
```diff
@@ -19,21 +19,29 @@ final class MockSiteSegmentsService: SiteSegmentsService {
+                           icon: URL(string: "https://s.w.org/style/images/about/WordPress-logotype-standard.png")!)
+    }
+
+    private func longSubtitle(id: String) -> SiteSegment {
```
I like the addition of this for testing! 👍

ctarda commented:
Oops, good catch @kwonye thanks!

I've pushed an update to bring the branch up to date with `develop` and to tint the icon. The icon of the Segment with a two line subtitle should be blue, while the others should be red, as declared in `MockSiteSegmentsService`

![simulator screen shot - iphone x - 2018-11-20 at 11 27 46](https://user-images.githubusercontent.com/2722505/48749838-9b17d200-ecb7-11e8-9594-52fe95353105.png)

kwonye commented:
Looks good!

SylvesterWilmott commented:
Looks good!

For each cell in the table there is a title and subtitle. It appears the title is larger than 17pt?

kwonye commented:
@SylvesterWilmott I'll have you be the final arbiter of whether we can merge this or not 👍

SylvesterWilmott commented:
Also just noticed, the subtitle text appears to be 12pt? In this case it should be 15pt.

You can find details about text sizes in zeplin by referencing the right hand sidebar while a text layer is selected.

ctarda commented:
Thanks @SylvesterWilmott . As with the previous PR I tried to approximate as much as possible. 

The Title is now styled with `body` as style and `semibold` as weight. The Subtitle is `callout` and `regular`

ctarda commented:
And I forgot to add a screenshot:

![simulator screen shot - iphone x - 2018-11-21 at 14 05 37](https://user-images.githubusercontent.com/2722505/48822054-96761b00-ed96-11e8-91a5-e53bc4458c1a.png)

SylvesterWilmott commented:
Thanks!

Text sizes look good

- Can we allow those table cells to expand with the text. The default state would be one line for the title and one line for the subtitle but as we add more lines the cells can change height:

<img width="406" alt="screenshot 2018-11-21 at 08 54 10" src="https://user-images.githubusercontent.com/36532468/48829783-08d10600-ed6b-11e8-9398-a31c3cf5f897.png">

- Can we change the text color in the cells to `darkGrey`

ctarda commented:
Thanks @SylvesterWilmott I think this is ready for another round. 

I made the cells support dynamic height. I double checked the test color and it is darkGrey, and it seems to match the RGB values declared on Zeplin. Please let me know if that's not the case, to take another look.

This is how this looks on an iPhone SE:
![simulator screen shot - iphone se - 2018-11-26 at 09 41 46](https://user-images.githubusercontent.com/2722505/48988333-ea4e7e80-f15f-11e8-91bd-6b6005945724.png)

This would be an iPhone X:
![simulator screen shot - iphone x - 2018-11-26 at 09 40 19](https://user-images.githubusercontent.com/2722505/48988339-f6d2d700-f15f-11e8-9a56-cc1c32645724.png)

And this would be on the new iPad Pro 11"
![simulator screen shot - ipad pro 11-inch - 2018-11-26 at 09 42 59](https://user-images.githubusercontent.com/2722505/48988351-094d1080-f160-11e8-888f-7b95169edc86.png)

@kwonye You might want to take a look at the last couple of commits, as I made some slight changes to the constraints attached to the header (I changed the width from being equal to the table view's to be lessThanOrEqualTo the table view's)

SylvesterWilmott commented:
Looks good!

- For the text it still appears as black in the screenshots
<img width="526" alt="screenshot 2018-11-26 at 09 16 53" src="https://user-images.githubusercontent.com/36532468/49004365-09e3a800-f15c-11e8-900d-3948c2791fc9.png">

- As mentioned in 10545: the tablet max-widths of the table should be 512pt

ctarda commented:
Thanks @SylvesterWilmott 

I've updated the width constraints for tablets (regular height and width). Here is how it looks on a 11" iPad Pro now (I have not scaled the screenshot):

![simulator screen shot - ipad pro 11-inch - 2018-11-27 at 12 37 23](https://user-images.githubusercontent.com/2722505/49059080-12aea980-f242-11e8-9607-4e0df87606f8.png)

I've double and triple checked the screenshot with Pixelmator and it seems like the color of the cells title and subtitle seems to be R: 46, G: 68, B: 83. Would you mind checking that again please?

<img width="274" alt="screenshot 2018-11-27 at 12 46 23 pm" src="https://user-images.githubusercontent.com/2722505/49059156-7e911200-f242-11e8-9fb3-d9b76affca3f.png">

SylvesterWilmott commented:
Thanks!

Layout looks great, colours are good.

Couple more minor things:

- The table row strokes should be `greyLighten20`
- Can we remove the cell stroke from the last child of the list?

ctarda commented:
Thanks!

I've updated the separator color and removed the one in the last cell.

![simulator screen shot - iphone x - 2018-11-27 at 18 57 55](https://user-images.githubusercontent.com/2722505/49077572-8ff41180-f276-11e8-8cbb-345f553602e3.png)

SylvesterWilmott commented:
One final thing (I didn't notice before) can we tint the chevron also to `greyLighten20`.

After that I think we're all good

ctarda commented:
Thanks @SylvesterWilmott I've updated the chevron.

![simulator screen shot - iphone x - 2018-11-28 at 09 29 52](https://user-images.githubusercontent.com/2722505/49122661-55c75600-f2f0-11e8-90bc-c2591637f652.png)

My checklist for the upcoming UI reviews would be:

- [ ] Width of lists on iPad
- [ ] Fonts and colours in cells
- [ ] Color of separator
- [ ] Hide separator for last cell
- [ ] Color of disclosure indicators

Thanks!

SylvesterWilmott commented:
Hmm the colour change of the separators didn't seem to make an effect. Otherwise everything looks good!

ctarda commented:
Sorry about that, it seems like I pushed the change to the wrong branch. I've updated it again.

![simulator screen shot - iphone x - 2018-11-28 at 17 32 46](https://user-images.githubusercontent.com/2722505/49142448-cb084a80-f333-11e8-9469-344dcb62be2d.png)
